### PR TITLE
Studio 1.2.0 Reference Cleanup

### PR DESCRIPTION
**Pull request by openbots-ff**
- Removed all references to OpenBots.Engine from all Command projects
- Resolved all tangled dependencies in OpenBots.UI and removed the project and its references from Studio/Commands
- Moved all Command Test projects to OpenBots.Commands.Test

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.NLG/OpenBots.Commands.NLG.csproj [hunk 1]
```diff
@@ -33,7 +33,7 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="SharpSimpleNLG, Version=1.2.1.0, Culture=neutral, processorArchitecture=MSIL">
+    <Reference Include="SharpSimpleNLG, Version=1.2.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.NLG/OpenBots.Commands.NLG.csproj:36]:
```suggestion
    <Reference Include="SharpSimpleNLG, Version=1.2.1.0, Culture=neutral, processorArchitecture=MSIL">
```

File: OpenBots.Commands/OpenBots.Commands.Task/RunTaskCommand.cs [hunk 2]
```diff
@@ -238,9 +237,9 @@ private void TaskPathControl_TextChanged(object sender, EventArgs e)
 			_passParameters.Checked = false;
 		}
 
-		private void PassParametersCheckbox_CheckedChanged(object sender, EventArgs e, IfrmCommandEditor editor)
+		private void PassParametersCheckbox_CheckedChanged(object sender, EventArgs e, IfrmCommandEditor editor, ICommandControls commandControls)
 		{
-			AutomationEngineInstance currentScriptEngine = new AutomationEngineInstance(null);
+			var currentScriptEngine = commandControls.CreateAutomationEngineInstance(null); //new AutomationEngineInstance(null);
```
Comment (openbots-nc) [inline OpenBots.Commands/OpenBots.Commands.Task/RunTaskCommand.cs:242]: Do you still need the commented code?

File: OpenBots.Commands/OpenBots.Commands.Task/RunTaskCommand.cs [hunk 3]
```diff
@@ -286,14 +285,14 @@ private void PassParametersCheckbox_CheckedChanged(object sender, EventArgs e, I
...
-			AutomationEngineInstance parentAutomationEngineInstance = (AutomationEngineInstance)sender;
+			var parentAutomationEngineInstance = (IAutomationEngineInstance)sender;
 			string childTaskPath = v_TaskPath.ConvertUserVariableToString(parentAutomationEngineInstance);
 			string parentTaskPath = parentAutomationEngineInstance.FileName;
 
 			//create variable list
 			InitializeVariableLists(parentAutomationEngineInstance);
 
-			var childAutomationEngineInstance = new AutomationEngineInstance((Logger)Log.Logger);
+			var childAutomationEngineInstance = parentAutomationEngineInstance.CreateAutomationEngineInstance((Logger)Log.Logger); //new AutomationEngineInstance((Logger)Log.Logger);
```
Comment (openbots-nc) [inline OpenBots.Commands/OpenBots.Commands.Task/RunTaskCommand.cs:295]: Do you still need this commented code?

**Review by openbots-nc — APPROVED** [review]

File: OpenBots.Commands/OpenBots.Commands.Task/RunTaskCommand.cs (hunk 3 above)
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Task/RunTaskCommand.cs]: removed

File: OpenBots.Commands/OpenBots.Commands.Task/RunTaskCommand.cs (hunk 2 above)
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Task/RunTaskCommand.cs]: removed

**Review by openbots-su — APPROVED** [review]

**Review by openbots-nc — APPROVED** [review]

**Pull request merged**